### PR TITLE
chore: adding encoding and txTypes fields to metadata

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

- Adding the `encoding` and `tx_types` fields specified in [cosmos/ibc #642](https://github.com/cosmos/ibc/pull/642/files) to interchain accounts `Metadata`
- Updating `Metadata` validation to support new fields
- Updating test cases

closes: #820

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: proto/ibc/applications/interchain_accounts/v1/metadata.proto [hunk 1]
```diff
@@ -18,4 +18,8 @@ message Metadata {
   // address defines the interchain account address to be fulfilled upon the OnChanOpenTry handshake step
   // NOTE: the address field is empty on the OnChanOpenInit handshake step
   string address = 4;
+  // encoding defines the supported codec format
+  string encoding = 5;
+  // tx_types defines the type of transactions the interchain account can execute
+  string tx_types = 6;
```
Comment (damiannolan) [inline proto/ibc/applications/interchain_accounts/v1/metadata.proto:24]: I defined as per the spec, but I think it should be `tx_type`

Comment (AdityaSripal) [inline proto/ibc/applications/interchain_accounts/v1/metadata.proto:24]: Yes, will change to `tx_type` in spec. cc: @colin-axner

**Review by AdityaSripal — APPROVED** [review]
ACK, pending review from @colin-axner

**Review by colin-axner — APPROVED** [review]
LGTM! Nice work!

**Review by seantking — APPROVED** [review]

**Review by AdityaSripal — APPROVED** [review]
Reapproved

**Pull request merged**